### DISCUSSION TITLE
How to add additional meta-data to templates?

### DISCUSSION THREAD

**Discussion by iburkhardt** [projectdiscovery/nuclei#941] [Q&A] [answered]
Hello everyone,

It is possible to put in the template information something like:

remediation effort: easy, hard etc.?
something like:
discoverability

info:
  name: Test HTTP Template
  author: pdteam
  Severity: info
  remediation effort: easy
  discoverability:

**Comment by ehsandeep** [comment, marked as answer]
@iburkhardt under `info:` block, you can define any number of `key: value` formatted data.

If you asking this in the context of a custom template: **Yes**, it's possible to add this information.
If you asking this in the context of existing public templates, this can not be done in automated way and & requires manual effort.

**Reply by forgedhallpass** [reply]
Note: we are making the template parsing stricter to improve template validation and to avoid typos etc., hence starting with the next release (probably `2.4.4`), the desired custom meta-data will be moved under `additional-fields` attribute.

**Comment by iburkhardt** [comment]
@ehsandeep That was my question, if I could add other information! Many thanks for the reply.

**Reply by ehsandeep** [reply]
np, here is an example as well:-

```yaml
info:
  name: Test HTTP Template
  author: pdteam
  severity: info
  complexity: low
  remediation: xx
  cvss: xx
  x: xx
  ```
